### PR TITLE
[core] Upgrade Node.js to v18 on CircleCI, CodeSandbox and Netlify

### PR DESCRIPTION
**Pull request by ZeeshanTamboli**
Similar to https://github.com/mui/material-ui/pull/37173.

### CODE REVIEW

File: packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx [hunk 1]
```diff
@@ -647,7 +647,7 @@ describe('<DateField /> - Editing', () => {
   describeAdapters('Pasting', DateField, ({ adapter, render, renderWithProps, clickOnInput }) => {
     const firePasteEvent = (input: HTMLInputElement, pastedValue: string) => {
       act(() => {
-        const clipboardEvent = new Event('paste', {
+        const clipboardEvent = new window.Event('paste', {
```
[the remark this answers is not in the document]
Comment (LukasTy) [reply packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx:650]: It's strange that numerous tests on Data Grid side using `new Event('scroll')` pass, but changing this test to `new Event('paste')` (without the second argument) still results in failures... 🤷

**Review by LukasTy — APPROVED** [review]
Thank you for looking at this matter and resolving the test failure! 💯 👍 
@flaviendelangle do you see any potential problems with the test change? 🤔

File: packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZeeshanTamboli) [reply packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx:650]: @LukasTy Those tests are not being run on JSDom.
```jsx
      if (isJSDOM) {
        // Need layouting
        this.skip();
      }
```

[the remark this answers is not in the document]
Comment (ZeeshanTamboli) [reply packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx:650]: Maybe you missed checking the `before` hook 😅 
```js
    before(function beforeHook() {
      if (isJSDOM) {
        // Need layouting
        this.skip();
      }
    });
```

**Review by cherniavskii — APPROVED** [review]
Thank you!

**Pull request merged**